hypergan/search/random_search.py

```diff
@@ -5,116 +5,82 @@
-            "clipped_d_weights": 0.01,
-            "clipped_gradients": False,
-            "d_decay": 0.995, "d_momentum": 1e-05,
-            "d_learn_rate": 0.001,
-            "g_decay": 0.995,
-            "g_momentum": 1e-06,
-            "g_learn_rate": 0.0005,
-        }
-
-        #trainers.append(hg.trainers.rmsprop_trainer.config(**rms_opts))
+        self.options = {**self.options, **overrides}
 
-        adam_opts = {}
-
-        adam_opts = {
-            'd_learn_rate': [1e-3,1e-4,5e-4,1e-2,1e-6],
-            'g_learn_rate': [1e-3,1e-4,5e-4,1e-2,1e-6],
-            'd_beta1': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'd_beta2': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'g_beta1': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'g_beta2': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'd_epsilon': [1e-8, 1, 0.1, 0.5],
-            'g_epsilon': [1e-8, 1, 0.1, 0.5],
-            'd_clipped_weights': [False, 0.01],
-            'clipped_gradients': [False, 0.01]
-        }
-
-        #trainers.append(hg.trainers.adam_trainer.config(**adam_opts))
+    def range(self):
+        return list(np.linspace(0, 1, num=1000000))
 
+    def trainers(self):
+        trainers = []
         any_opts = {}
 
         tftrainers = [
                 tf.train.AdadeltaOptimizer,
                 tf.train.AdagradOptimizer,
                 tf.train.GradientDescentOptimizer,
                 tf.train.AdamOptimizer,
-
+                tf.train.AdagradDAOptimizer, # TODO missing param
+                tf.train.MomentumOptimizer,
+                tf.train.ProximalGradientDescentOptimizer,
+                #tf.train.FtrlOptimizer,
+                tf.train.RMSPropOptimizer,
+                tf.train.ProximalAdagradOptimizer,
         ]
-        # TODO FtrlOptimizer
-        # TODO ProximalAdagradOptimizer
-        # TODO ProximalGradientDescentOptimizer
 
-        any_opts = {
-            'd_learn_rate': [1e-3,1e-4,5e-4,1e-2,1e-6],
-            'g_learn_rate': [1e-3,1e-4,5e-4,1e-2,1e-6],
-            'd_beta1': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'd_beta2': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'g_beta1': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'g_beta2': [0.9, 0.99, 0.999, 0.1, 0.01, 0.2, 1e-8],
-            'd_epsilon': [1e-8, 1, 0.1, 0.5],
-            'g_epsilon': [1e-8, 1, 0.1, 0.5],
-            'g_momentum': [0,0.1,0.01,1e-6,1e-5,1e-1,0.9,0.999, 0.5],
-            'd_momentum': [0,0.1,0.01,1e-6,1e-5,1e-1,0.9,0.999, 0.5],
-            'd_decay': [0.8, 0.9, 0.99,0.999,0.995,0.9999,1],
-            'g_decay': [0.8, 0.9, 0.99,0.999,0.995,0.9999,1],
-            'd_rho': [0.99,0.9,0.95,0.1,0.01,0],
-            'g_rho': [0.99,0.9,0.95,0.1,0.01,0],
-            'd_initial_accumulator_value': [0.99,0.9,0.95,0.1,0.01],
-            'g_initial_accumulator_value': [0.99,0.9,0.95,0.1,0.01],
+        selector = hc.Selector({
+            'd_learn_rate': self.range(),
```

Comment:
I wonder if we want the range to be smaller on learn rate.  Something like self.range / 100.0